File: Jan/Week 3/LeetCode/problem_2.py
# Day 16 Problem No 33

# 1768. Merge Strings Alternately => https://leetcode.com/problems/merge-strings-alternately/description/?envType=study-plan-v2&envId=leetcode-75

class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        merged = []
        i = 0
        j = 0
        while i < len(word1) or j < len(word2):
            if i < len(word1):
               merged.append(word1[i])
               i+=1
            if j < len(word2):
                merged.append(word2[j])
                j+=1
        return ''.join(merged)

File: Jan/Week 1/LeetCode/problem_2.cpp
// Day 2 Problem No 3

// 141. Linked List Cycle => https://leetcode.com/problems/linked-list-cycle/description/

class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(head==NULL) return false;
        map<ListNode*,bool> v;
        ListNode* tmp = head;
        while(tmp!=NULL){
            if(v[tmp]==true) return true;
            v[tmp] = true;
            tmp = tmp->next;
        }
        return false;
    }
};